pkg/controllermanager/controller/shoot/shoot_control.go

```diff
@@ -143,46 +113,33 @@ func (c *Controller) reconcileShootKey(key string) error {
 	// Ignore Shoots which do not have the gardener finalizer.
 	if shoot.DeletionTimestamp != nil && !sets.NewString(shoot.Finalizers...).Has(gardenv1beta1.GardenerName) {
 		shootLogger.Debug("Do not need to do anything as the Shoot does not have my finalizer")
-		c.scheduler.Delete(shootID)
 		c.getShootQueue(shoot).Forget(key)
 		return nil
 	}
 
-	shootElement, err := c.newShootElement(shoot)
-	if err != nil {
-		return err
+	seedName := shoot.Spec.Cloud.Seed
+	if seedName == nil {
+		return fmt.Errorf("shoot has not yet been scheduled on a seed")
 	}
 
-	var (
-		mayReconcile bool
-		reason       *reconcilescheduler.Reason
-	)
+	seed, err := c.seedLister.Get(*seedName)
+	if err != nil {
+		return fmt.Errorf("could not find seed %s", *seedName)
+	}
 
 	// Check whether the shoot may be reconciled. We always allow reconciliation if the shoot shall be deleted or
 	// if the shoot shall be created.
-	switch {
-	case shoot.DeletionTimestamp != nil:
-		c.scheduler.Delete(shootID)
-		mayReconcile, reason = true, reconcilescheduler.NewReason(reconcilescheduler.CodeOther, "shoot shall be deleted")
-	case shoot.Spec.Cloud.Seed == nil:
-		shootLogger.Info("Skipping Shoot as it is not scheduled to a seed yet")
-		mayReconcile, reason = false, reconcilescheduler.NewReason(reconcilescheduler.CodeOther, "waiting for shoot to be scheduled to seed")
-	case shoot.Status.LastOperation != nil && shoot.Status.LastOperation.Type == gardencorev1alpha1.LastOperationTypeCreate:
-		mayReconcile, reason = true, reconcilescheduler.NewReason(reconcilescheduler.CodeOther, "shoot shall be created")
-	default:
-		mayReconcile, reason = c.scheduler.TestAndActivate(shootElement, shoot.Generation != shoot.Status.ObservedGeneration, shootIsSeed(shoot))
-	}
+	reason := health.CheckSeed(seed, c.identity)
 
 	startReconciliation := false
 
 	switch {
 	case mustIgnoreShoot(shoot.Annotations, c.config.Controllers.Shoot.RespectSyncPeriodOverwrite):
 		// Check whether the shoot has been marked as "never reconcile".
 		shootLogger.Info("Skipping reconciliation because Shoot is marked as 'to-be-ignored'.")
-
-	case !mayReconcile:
+	case reason != nil:
 		// If the shoot may not be reconciled (due to above decision) then we mark it as pending in case it was not failed
-		message := fmt.Sprintf("May not yet reconcile shoot %q: %s", shoot.Name, reason)
+		message := fmt.Sprintf("Cannot reconcile shoot %q: %s", shoot.Name, reason.Error())
```

Comment:
Can you please improve the error message? Now I see `'Cannot reconcile shoot "my-shoot": observed generation outdated (0/6)'`. Can we have something better like "Waiting for seed cluster to be up to date" or something similar?